pkg/lib/dialogs.jsx

```diff
@@ -82,19 +82,53 @@
  *   }
  * }
  *
+ * If you need to wait on the result of a Dialog before you can await "Dialogs.show"
+ *
+ * class Example extends React.Component {
+ *   static contextType = DialogsContext;
+ *
+ *   async function handleClick() {
+ *     try {
+ *       const result = await Dialogs.show(<MyDialog />);
+ *       console.log(result);
+ *     catch (err) {
+ *     }
+ *
+ *   }
+ *
+ *   function render() {
+ *     const Dialogs = this.context;
+ *     return <Button onClick={() => this.handleClick}>Open dialog</Button>;
+ *   }
+ * }
+ *
+ * class MyDialog extends React.Component {
+ *   static contextType = DialogsContext;
+ *
+ *   render() {
+ *     <Button onClick={() => Dialogs.close("yes")}>Yes</Button>
+ *     <Button onClick={() => Dialogs.close("no")}>No</Button>
+ *   }
+ * }
+ *
  *
  * - Dialogs.show(component)
  *
  * Calling "Dialogs.show" will render the given component as a direct
  * child of the inner-most enclosing "WithDialogs" component.  The
  * component is of course intended to be a dialog, such as
  * Patternfly's "Modal".  There is only ever one of these; a second
- * call to "show" will remove the previously rendered component.
- * Passing "null" will remove the currently rendered componenet, if any.
+ * call to "show" is considered a bug and "Dialogs.close" should be called first.
```

Comment:
We should say here that show returns a promise that is settled by either close or reject_close.

Reply:
Good catch!